src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadonlyStructTests.cs

```diff
@@ -1308,6 +1308,211 @@ public static void Main()
             CompileAndVerify(comp, expectedOutput: "SpanOpCalled", verify: Verification.Fails);
         }
 
+        [Fact]
+        public void ReadOnlyMembers_Metadata()
+        {
+            var csharp = @"
+public struct S
+{
+    public void M1() {}
+    public readonly void M2() {}
+
+    public int P1 { get; set; }
+    public readonly int P2 => 42;
+    public int P3 { readonly get => 123; }
+    public int P4 { readonly set {} }
```

Comment:
>public int P4 { readonly set {} } [](start = 4, length = 33)

Can you also test a readonly getter in a readonly property? (assuming we indeed want to allow that)

Reply:
This is disallowed by the spec in order to be consistent with the behavior around visibility modifiers (e.g. `public int P { public get; }` is not allowed). The check is coming in a future PR.

Reply:
Added a note to test plan

---
In reply to: [267478117](https://github.com/dotnet/roslyn/pull/34260#discussion_r267478117) [](ancestors = 267478117)